cassandra/cluster.py

```diff
@@ -4571,7 +4653,9 @@ def _query(self, host, message=None, cb=None):
         connection = None
         try:
             # TODO get connectTimeout from cluster settings
-            connection, request_id = pool.borrow_connection(timeout=2.0, routing_key=self.query.routing_key if self.query else None)
+            connection, request_id = pool.borrow_connection(timeout=2.0, routing_key=self.query.routing_key if self.query else None, 
+                                                            keyspace=self.query.keyspace if self.query else None, 
+                                                            table=self.query.table if self.query else None)
```

Comment:
At this point it would probably be cleaner to make separate definitions based on `self.query` being None or not. I mean something like:
```python
if self.query is None:
    connection, request_id = pool.borrow_connection(...)
else:
    connection, request_id = pool.borrow_connection(...)
```

Reply:
Done